cmd/svcat/broker/register_cmd.go

```diff
@@ -18,57 +18,122 @@ package broker
 
 import (
 	"fmt"
+	"os"
+	"strings"
+	"time"
 
 	"github.com/kubernetes-incubator/service-catalog/cmd/svcat/command"
 	"github.com/kubernetes-incubator/service-catalog/cmd/svcat/output"
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	servicecatalog "github.com/kubernetes-incubator/service-catalog/pkg/svcat/service-catalog"
 	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // RegisterCmd contains the information needed to register a broker
 type RegisterCmd struct {
-	BrokerName string
-	Context    *command.Context
-	URL        string
+	*command.Waitable
+
+	Context *command.Context
+
+	BasicSecret       string
+	BearerSecret      string
+	BrokerName        string
+	CAFile            string
+	ClassRestrictions []string
+	PlanRestrictions  []string
+	SkipTLS           bool
+	RelistBehavior    string
+	RelistDuration    time.Duration
+	URL               string
 }
 
 // NewRegisterCmd builds a "svcat register" command
 func NewRegisterCmd(cxt *command.Context) *cobra.Command {
 	registerCmd := &RegisterCmd{
-		Context: cxt,
+		Context:  cxt,
+		Waitable: command.NewWaitable(),
 	}
 	cmd := &cobra.Command{
-		Use:   "register NAME --url URL",
+		Use:   "register NAME URL",
 		Short: "Registers a new broker with service catalog",
 		Example: command.NormalizeExamples(`
-		svcat register mysqlbroker --url http://mysqlbroker.com
+		svcat register mysqlbroker http://mysqlbroker.com
 		`),
 		PreRunE: command.PreRunE(registerCmd),
 		RunE:    command.RunE(registerCmd),
 	}
-	cmd.Flags().StringVar(&registerCmd.URL, "url", "",
-		"The broker URL (Required)")
-	cmd.MarkFlagRequired("url")
+	cmd.Flags().StringVar(&registerCmd.BasicSecret, "basic-secret", "",
```

Comment:
Do you know how the secrets work when we are making a ClusterServiceBroker vs a ServiceBroker. For a namespaced broker, I assume that the secret is in the same namespace. For a cluster-scoped broker though, do they have to provide a namespace? Or is it just embedded in the format, e.g. "namespace/secret-name"?

Reply:
Currently it tags the secret as being in whatever namespace the command was executed in. This might be a problem when we want to change the command to create namespaced broker.